tests/lib/desktop-portal.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+#shellcheck source=tests/lib/pkgdb.sh
+. "$TESTSLIB"/pkgdb.sh
+
+TEST_UID="$(id -u test)"
+USER_RUNTIME_DIR="/run/user/${TEST_UID}"
+
+setup_portals() {
+    # Clean up any stale configuration
+    teardown_portals
+
+    # Install dependencies and configure service activation for fake
+    # portal UI.
+    distro_install_package python3-dbus python3-gi xdg-desktop-portal
```

Comment:
Sorry for the long delay in responding to this.  I tried moving the install of xdg-desktop-portal to `pkg_dependencies`, but couldn't get it to work.  Various pieces of state end up leaking from one test to the next.  It seems some tests end up leaving a stale document portal mount behind which can interfere with future tests.  I tried inspecting tests to try and find culprits, but gave up after chasing many of them.

So instead I've kept the install of xdg-desktop-portal in this script, but moved the Python library installs to `pkg_dependencies`.  One other alternative might be to install xdg-desktop-portal, and then move the D-Bus service activation files out of the way so we only make portals visible in tests that expect it.

Reply:
This seems fine (and pragmatic) to me, we should probably add a TODO/FIXME or similar (this comment is probably fine in there) so that we remember the rational for this unusual setup. If @sergiocazzolato is also happy I think this can go in (thanks for the test improvements).